src/zenml/zen_server/routers/schedule_endpoints.py

```diff
@@ -193,15 +193,18 @@ def update_schedule(
 @async_fastapi_endpoint_wrapper
 def delete_schedule(
     schedule_id: UUID,
+    soft: bool,
     _: AuthContext = Security(authorize),
 ) -> None:
     """Deletes a specific schedule using its unique id.
 
     Args:
         schedule_id: ID of the schedule to delete.
+        soft: Soft deletion will archive the schedule.
     """
```

Comment:
Do you think we need different RBAC permissions for archiving and hard deletion?